test/performance/io/format_fasta_benchmark.cpp

```diff
@@ -0,0 +1,178 @@
+#include <seqan3/io/sequence_file/input_format_concept.hpp>
+#include <seqan3/io/sequence_file/output.hpp>
+#include <seqan3/io/sequence_file/output_format_concept.hpp>
+#include <seqan3/io/sequence_file/format_fasta.hpp>
+#include <seqan3/range/view/convert.hpp>
+
+#include <sstream>
+
+using namespace seqan3;
+
+static void write3(benchmark::State& state)
+{
+    std::stringstream ostream;
+    sequence_file_format_fasta format;
+    sequence_file_output_options options;
+    std::string id{"seq"};
+    dna5_vector seq{"ACTAGACTAGCTACGATCAGCTACGATCAGCTACGA"_dna5};
+
+    for (auto _ : state)
+    {
+        format.write(ostream, options, seq, id, std::ignore);
+    }
+}
+
+BENCHMARK(write3);
+
+#if __has_include(<seqan/seq_io.h>)
+
+    static void write2(benchmark::State& state)
+    {
+        std::stringstream outStream;
+        seqan::CharString meta = "seq";
+        seqan::Dna5String seq = "ACTAGACTAGCTACGATCAGCTACGATCAGCTACGA";
+
+        for (auto _ : state)
+        {
+            seqan::writeRecord(outStream, meta, seq, seqan::Fasta());
+        }
+    }
+
+    BENCHMARK(write2);
+#endif
+
+static void read3(benchmark::State& state)
+{
+    std::string dummy_file{};
+    for (size_t idx = 0; idx < 10000000; idx++)
+        dummy_file += ">seq\nACTAGACTAGCTACGATCAGCTACGATCAGCTACGA\n";
+    std::stringstream istream{dummy_file};
+    sequence_file_format_fasta format;
+    sequence_file_input_options<dna5, false> options;
+    std::string id;
+    dna5_vector seq;
+    for (auto _ : state)
+    {
+        format.read(istream, options, seq, id, std::ignore);
+	id.clear();
+	seq.clear();
+    }
+}
+BENCHMARK(read3);
+
+static void read3_from_file(benchmark::State& state)
+{
+    // Create fasta file
+    std::string filename{"Test_Format_Fasta_Benchmark.fasta"};
+    sequence_file_output fout{filename};
+    for (size_t idx = 0; idx < 10000000; idx++)
+    {
+        std::string id{ "seq" };
+        dna5_vector seq{ "ACTAGACTAGCTACGATCAGCTACGATCAGCTACGA"_dna5 };
+        fout.push_back(std::tie(seq, id));   // as a tuple
+    }
+
+    sequence_file_input fin{filename};
+    auto file_it = fin.begin();
+    std::string id;
+    dna5_vector seq;
+
+    for (auto _ : state)
+    {
+        [[maybe_unused]] volatile auto record = *file_it;
+        ++file_it;
+    }
+}
+
+BENCHMARK(read3_from_file);
+
+
+#if __has_include(<seqan/seq_io.h>)
+
+#include <fstream>
+
+static void read2(benchmark::State& state)
+    {
+        seqan::CharString meta;
+        seqan::Dna5String seq;
+        std::string dummy_file{};
+        for (size_t idx = 0; idx < 10000000; idx++)
+            dummy_file += ">seq\nACTAGACTAGCTACGATCAGCTACGATCAGCTACGA\n";
+        std::istringstream istream{dummy_file};
+
+        seqan::VirtualStream<char, seqan::Input> comp;
+        open(comp, istream);
+        auto it = seqan::directionIterator(comp, seqan::Input());
+
+        for (auto _ : state)
+        {
+            readRecord(meta, seq,  it, seqan::Fasta{});
```

Comment:
```suggestion
            [[maybe_unused]] readRecord(meta, seq,  it, seqan::Fasta{});
```
add this to eevery read line just to make sure

Reply:
@smehringer  I get an error message, if I do that because I am not defining record here, so should I use this in front of the definitions of meta, seq and it/reader?

Reply:
ah I see. Then don't add it here.